MvvmHelpers/ObservableRangeCollection.cs

```diff
@@ -46,12 +46,19 @@ public void AddRange(IEnumerable<T> collection, NotifyCollectionChangedAction no
 
             if (notificationMode == NotifyCollectionChangedAction.Reset)
             {
+                bool raiseEvents = false;
                 foreach (var i in collection)
+                {
                     Items.Add(i);
+                    raiseEvents = true;
+                }
 
-                OnPropertyChanged(new PropertyChangedEventArgs("Count"));
-                OnPropertyChanged(new PropertyChangedEventArgs("Item[]"));
-                OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
+                if (raiseEvents)
```

Comment:
couldn't we just check the count of collection? instead of having a bool here?

Reply:
The collection may not have a count.